File: 2022/julia/day1.jl
function elves_total(lines)
    elves = Array{Int32}[[]]

    i = 1
    calories = tryparse.(Int32, lines)
    for c in calories
        if isnothing(c)
            i += 1
            push!(elves, [])
            continue
        end
        push!(elves[i], c)
    end
    elves_total = sum.(elves)
end

function sol1(lines)
    calories = elves_total(lines)
    maximum(calories)
end

function sol2(lines)
    calories = elves_total(lines)
    sum(sort(calories, rev=true)[1:3])
end

readlines("../input/1.txt") |> sol1 |> println
readlines("../input/1.txt") |> sol2 |> println

File: 2022/haskell/src/Day1.hs
module Day1 where
import Data.List
import qualified Data.Text as T
import qualified Data.Text.Read as T
import qualified Data.Text.IO as T

sortedSums contents =
  reverse $ sort sums
  where
    sums = sum <$> partInts
    partInts = fmap (read . T.unpack) <$> partLines :: [[Int]]
    partLines = T.lines <$> parts
    parts = T.splitOn "\n\n" contents

sol1 = head . sortedSums

sol2  = sum . take 3 . sortedSums
